src/shogun/mathematics/linalg/ratapprox/logdet/LogDetEstimator.cpp

```diff
@@ -125,23 +125,24 @@ SGVector<float64_t> CLogDetEstimator::sample(index_t num_estimates)
 	index_t num_trace_samples=m_trace_sampler->get_num_samples();
 	SGVector<float64_t> samples(num_estimates);
 	samples.zero();
+	float64_t result=0.0;
 
+#pragma omp parallel for reduction(+:result)
 	for (index_t i = 0; i < num_estimates; ++i)
 	{
+		result=0.0;
 		for (index_t j = 0; j < num_trace_samples; ++j)
 		{
 			SG_INFO(
 				"Computing log-determinant trace sample %d/%d\n", j,
 				num_trace_samples);
 			// get the trace sampler vector
 			SGVector<float64_t> s = m_trace_sampler->sample(j);
-			// calculate the result for sample s
-			float64_t result = m_operator_log->compute(s);
-			{
-				samples[i] += result;
-			}
-		}
+			// calculate the result for sample s and add it to previous
+			result += m_operator_log->compute(s);
```

Comment:
just another question.
Is this `compute` method threadsafe, i.e. `const`
If not, we need to make it so to avoid the race condition analyser moaning

Reply:
Actually, all of the methods called in this parallel loop should be const

Reply:
I am having trouble in using const in ```compute``` due to [this](https://github.com/shogun-toolbox/shogun/blob/32a6746251e4864269feb08702786ed1214be0b6/src/shogun/mathematics/linalg/ratapprox/logdet/opfunc/LogRationalApproximationCGM.cpp#L64) line..
The other method i.e. ```sample(j)``` is thread safe.